content/src/content/jcr_root/apps/core/wcm/sandbox/components/teaser/v1/teaser/teaser.html

```diff
@@ -17,17 +17,13 @@
      data-sly-use.templates="core/wcm/components/commons/v1/templates.html"
      data-sly-test.hasContent="${teaser.imageResource}"
      class="cmp-teaser${!wcmmode.disabled ? ' cq-dd-image' : ''}">
-
     <div class="cmp-teaser__image" data-sly-resource="${teaser.imageResource}"></div>
-    <sly data-sly-test.withCTA="${teaser.withCTA}" data-sly-list="${teaser.CTAs}">
-        <a class="cmp-teaser__call-to-action-link" href="${item.path}">${item.title}</a>
-    </sly>
-
-    <h2 class="cmp-teaser__title" data-sly-test.title="${teaser.title}">
-        <a class="cmp-teaser__title-link" href="${teaser.linkURL}" data-sly-unwrap="${!teaser.linkURL || withCTA || teaser.hideTitleLink}">${title}</a>
+    <h2 class="cmp-teaser__title" data-sly-test.title="${teaser.title}" data-sly-element="${teaser.titleType}">
+        <a class="cmp-teaser__title-link" href="${teaser.linkURL}" data-sly-unwrap="${!teaser.linkURL || teaser.hideTitleLink}">${title}</a>
     </h2>
-    <p class="cmp-teaser__description" data-sly-test.description="${teaser.description}">
-        <a class="cmp-teaser__description-link" href="${teaser.linkURL}" data-sly-unwrap="${!teaser.linkURL || withCTA || teaser.hideDescriptionLink}">${description}</a>
-    </p>
+    <div class="cmp-teaser__description" data-sly-test.description="${teaser.description}">${teaser.description @ context="html"}</div>
+    <div class="cmp-teaser__call-to-action-container" data-sly-test.withCTA="${teaser.withCTA}" data-sly-list="${teaser.CTAs}">
+        <a class="cmp-teaser__call-to-action-link" href="${item.path}">${item.title}</a>
```

Comment:
I wonder if we should split this template up a little to enable easier override of the sub elements. It might be common for example to want to provide a different action markup to include e.g. an icon.

Reply:
I agree. How granular do you think we should be here?

Reply:
I would be tempted to go fully granular:
```
teaser.html
image.html
title.html
description.html
actions.html
action.html
```

So, an override of `teaser.html` would then simply be about reordering the constituent parts (with may be common in order to achieve a particular layouts which may not be easily achieved with CSS alone) or to introduce some additional grouping (e.g. title and description under a "copy" element).